backend/src/layer/data/flickr_api/api_request.rs

```diff
@@ -0,0 +1,215 @@
+        debug!("request_err finished: {:?}", res);
+        println!("PRINT VALUE: {:?}", res);
+        Ok(res)
+    }
+
+    /// This method creates an api request url.
+    /// This url is used to create an request with the [`request_sizes`] function.
+    /// # Errors
+    /// If the request could not be decoded ([`ImageHosterError::DecodeFailed`],
+    /// another request with [`request_err`] will be attempted.
+    /// This error request returns a more detailed error information.
+    /// # Returns
+    /// An Error (as above mentioned) or an [`ImageMetaData`] struct containing information about the requested image.
+    pub async fn flickr_photos_get_sizes(
+        &self,
+        photo_id: &str,
+    ) -> Result<ImageMetaData, ImageHosterError> {
+        let url = &format!(
+            "{BASE_URL}{GET_SIZES}{TAG_API_KEY}{api_key}{TAG_PHOTO_ID}{photo_id}{JSON}",
+            BASE_URL = Self::BASE_URL,
+            GET_SIZES = Self::GET_SIZES,
+            TAG_API_KEY = Self::TAG_API_KEY,
+            api_key = self.api_key,
+            TAG_PHOTO_ID = Self::TAG_PHOTO_ID,
+            JSON = Self::FORMAT
+        );
+        println!("flickr.photos.getSizes URL: {}", url); // TODO remove
+        match self.request_sizes(&url).await {
+            Ok(sizes) => Ok(self.parser.parse_get_sizes(sizes, photo_id)),
+            Err(e) => Err(self.determine_error(url, e).await)
+        }
+    }
+
+    /// This method creates an api request url.
+    /// This url is used to create an request with the [`request_license`] function.
+    /// # Errors
+    /// If the request could not be decoded ([`ImageHosterError::DecodeFailed`],
+    /// another request with [`request_err`] will be attempted.
+    /// This error request returns a more detailed error information.
+    /// # Returns
+    /// An Error (as above mentioned).
+    /// True if the given image is hosted under a valid license (see [`json_parser::get_valid_licences`] for more info).
+    /// False if not.
+    pub async fn flickr_photos_licenses_get_license_history(
+        &self,
+        photo_id: &str,
+    ) -> Result<bool, ImageHosterError> {
+        let url = &format!(
+            "{BASE_URL}{GET_LICENCE_HISTORY}{TAG_API_KEY}{api_key}{TAG_PHOTO_ID}{photo_id}{JSON}",
+            BASE_URL = Self::BASE_URL,
+            GET_LICENCE_HISTORY = Self::GET_LICENCE_HISTORY,
+            TAG_API_KEY = Self::TAG_API_KEY,
+            api_key = self.api_key,
+            TAG_PHOTO_ID = Self::TAG_PHOTO_ID,
+            JSON = Self::FORMAT
+        );
+        println!("flickr.photos.licenses.getLicenseHistory URL: {}", url); // TODO remove
+        match self.request_license(&url).await {
+            Ok(licenses) => Ok(self.parser.check_license(licenses)),
+            Err(e) => Err(self.determine_error(url, e).await)
+        }
+    }
+
+    /// This method creates an api request url.
```

Comment:
??? 
This method does NOT create the url. It requests data from the api (at that url). 
I think it would help if you sometimes rethingk your terminology 😕 .